clusterversion,*: remove most 19.2 and 20.1 cluster versions

thoszhang opened the pull request:
This PR removes most 19.2 and 20.1 cluster versions and their version gates or associated baked-in migrations. I left a few of them alone where it wasn't obvious what to do, and will open separate issues for those.

Related to #47447.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56397)
<!-- Reviewable:end -->

Commit:
sql/logictest: delete local-mixed-19.2-20.1 configuration

Release note: None

Commit:
clusterversion,sql: remove VersionSavepoints

This was an obsolete cluster version from 20.1.

Release note: None

Commit:
clusterversion,kvserver: remove VersionAtomicChangeReplicas{,Trigger}

These were obsolete cluster versions from 19.2.

Release note: None

Commit:
clusterversion,sqlmigrations: remove versions for baked-in migrations

This commit removes the following cluster versions:

- Version19_2
- VersionProtectedTimestamps
- VersionCreateRolePrivilege
- VersionStatementDiagnosticsSystemTables
- VersionSchemaChangeJob

Release note: None

Commit:
clusterversion,kv/bulk: remove VersionStart20_1

This cluster version was being used as the version gate for the
SST-based writer.

Release note: None

Commit:
clusterversion: remove Version20_1

This was an obsolete and unused cluster version.

Release note: None

Commit:
clusterversion,kvserver: remove VersionChangeReplicasDemotion

This was an obsolete cluster version from 20.1.

Release note: None

Commit:
clusterversion,importccl: remove VersionPartitionedBackup

This was an obsolete cluster version from 19.2.

Release note: None

Commit:
clusterversion,sql: remove 20.1 version gates

This commit removes the following cluster versions:

- VersionSecondaryIndexColumnFamilies
- VersionPrimaryKeyChanges
- VersionPrimaryKeyColumnsOutOfFamilyZero
- VersionNoExplicitForeignKeyIndexIDs
- VersionHashShardedIndexes

Release note: None

tbg reviewed (approved):
> Reviewed 3 of 3 files at r1, 7 of 7 files at r2, 3 of 3 files at r3, 4 of 4 files at r4, 3 of 3 files at r5, 6 of 6 files at r6, 3 of 3 files at r7, 2 of 2 files at r8, 3 of 3 files at r9.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56397)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani, @dt, and @knz)



<!-- Sent from Reviewable.io -->

knz reviewed (approved):
> Reviewed 3 of 3 files at r1, 4 of 7 files at r2, 1 of 3 files at r3, 2 of 4 files at r4, 1 of 3 files at r5, 4 of 6 files at r6, 1 of 3 files at r7, 3 of 3 files at r9.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56397)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani and @dt)



<!-- Sent from Reviewable.io -->

lucy-zhang commented:
TFTRs

bors r+
